src/Scalar.jl

```diff
@@ -0,0 +1,20 @@
+"""
+    Scalar{T}(x::T)
+
+Construct a statically-sized 0-dimensional array that contains a single element,
+`x`. This type is particularly useful for influencing broadcasting operations.
+"""
+immutable Scalar{T} <: StaticArray{T,0}
+    data::T
+end
+
+@inline (::Type{Scalar}){T}(x::Tuple{T}) = Scalar{T}(x[1])
```

Comment:
The StaticArrays interface requires a constructor that accepts a Tuple. I may have messed this up a little though. 

(PS - side effect is small - it only does this for a one-tuple which is scalar-like anyway).